### DISCUSSION TITLE
Max Bending Moment

### DISCUSSION THREAD

**Discussion by petewil1291** [sympy/sympy#25952] [General]
I'm using the beam module and for some reason the max_bmoment() function returns `(Interval(0, 6), 0)`

code:
```
from sympy.physics.continuum_mechanics.beam import Beam
from sympy import symbols
E, I = symbols("E I")
R1, R2 = symbols("R1 R2")
my_beam = Beam(6, E, I)
my_beam.apply_load(100, 0, 0, 6)
my_beam.apply_load(R1, 0,  -1)
my_beam.apply_load(R2, 6, -1)
my_beam.bc_deflection =[(0, 0), (6, 0)]
my_beam.solve_for_reaction_loads(R1, R2)
my_beam.bc_deflection=[(0,0), (6,0)]
my_beam.reaction_loads
```
The reactions returned are correct and then shear force and bending moment diagrams also look correct. The correct answer should be 450 at coordinate 3.

**Comment by moorepants** [comment]
The `max_bmoment()` function likely has a bug.

**Reply by petewil1291** [reply]
It looks like max_shear_force() works correctly. I dont understand through the code for those functions though.
Interestingly if I leave all of the inputs in terms of symbols, the max_bmoment() function returns the correct expression and the correct coordinate.
